### PR TITLE
Fix Quick unable to find module name when bundle name contains `-` #534

### PR DESCRIPTION
**Pull request by MP0w**
The PR should summarize what was changed and why. Here are some questions to
help you if you're not sure:

 - What behavior was changed? Running test by selecting in the Xcode test navigator is now working for test bundles containing dashes in the name
 - What code was refactored / updated to support this change? -
 - What issues are related to this PR? Or why was this change introduced? #534 #557 #651 

Checklist - While not every PR needs it, new features should consider this list:

 - [x] Does this have tests? Yes [See comment](https://github.com/Quick/Quick/pull/652#issuecomment-265172849)
 - [x] Does this have documentation? Yes
 - [x] Does this break the public API (Requires major version bump)? No
 - [x] Is this a new feature (Requires minor version bump)? No

Note: #557 fixed the issue with spaces but apparently there are other character causing troubles. Hopefully this are the only two commonly used in bundle names.

### CODE REVIEW

File: Sources/Quick/NSBundle+CurrentTestBundle.swift [hunk 1]
```diff
@@ -11,6 +11,16 @@ extension Bundle {
...
+    /**
+     Return the module name of the bundle.
+     Uses the bundle filename and transform it to match Xcode's transformation.
+     Module name has to be a valid "C99 extended identifier".
+     */
+    internal var moduleName: String {
+        return bundleURL.fileName
+            .replacingOccurrences(of: " ", with: "_")
+            .replacingOccurrences(of: "-", with: "_")
+    }
```
Comment (modocache) [inline Sources/Quick/NSBundle+CurrentTestBundle.swift]: Can you reuse [the same mechanism as the one we use for test names](https://github.com/Quick/Quick/blob/0fdf273c7c1463a5f25c3942587452b1e892918c/Sources/QuickObjectiveC/NSString%2BQCKSelectorName.m#L5-L34) here? Or is there a reason we can't?

Comment (MP0w) [inline Sources/Quick/NSBundle+CurrentTestBundle.swift]: oh, I didn't know about that category.... looks like good for module name as well.
Should I rename it to `String.c99ExtendedIdentifier` (assuming it is)

**Comment by MP0w** [comment]
@modocache adding another bundle can be a way but then we should do the same for space and we quickly end up with many targets I guess. Do we want this?
I wouldn't test if test were performed (I guess should be tested somewhere else?).
A unit test for this issue should check that a (swift) class defined in a bundle that contains a space/dash in the name can be retrieved at runtime (module name is correct).

File: Sources/Quick/NSBundle+CurrentTestBundle.swift (hunk 1 above)
Comment (jeffh) [inline Sources/Quick/NSBundle+CurrentTestBundle.swift]: Renaming + reusing that method sounds great.

**Comment by jeffh** [comment]
I'm ok with adding another test bundle that has many invalid characters. I think we don't want to regress from it.

**Comment by MP0w** [comment]
I pushed some changes to use `c99ExtendedIdentifier`. I converted it to swift because we can't use a riding header.
For the test I currently test the behaviour with the existing test bundle that contains a `-`.
I tried to add another test bundle but I'm not exactly sure how I should do it, If we want to add a test in this new test target which contains special characters then we would have to create 3 targets right? (iOS, OSX, tvOS).
Would it be enough to test it in the current targets + test that `c99ExtendedIdentifier` works also with spaces, etc ?

**Comment by jeffh** [comment]
Thanks @MP0w, I've merged the current work as is. I would ok with just one extra test bundle on iOS that has that if you have time. Otherwise, it'll be part of the work again if a regression occurs.

**Pull request merged**